test/es-module/test-esm-pkgname.mjs

```diff
@@ -0,0 +1,18 @@
+// Flags: --experimental-modules
```

Comment:
Since this gets missed now, it may be good to add a test for exports that ensures that they aren't applied for exactly the case of `../hidden.js` even if the parent directory happens to have a `package.json` file that doesn't export `hidden.js`.

Reply:
I'm not sure I follow this point or how it relates to this PR, can you clarify which test file changes you are referring to?

Reply:
Not necessary as part of this PR. More a "oops, guess we never had a regression test covering this".

Reply:
I still don't understand the exact regression / issue?

Reply:
We started treating `../` as a package name and would've loaded the export map etc. (I assume) but none of our tests broke. At least that's my current understanding. In other words: We don't have the following test right now:

```js
// test/fixtures/node_modules/pkg-exports/lib/a.mjs
import '../hidden.js'; // should succeed but would be broken if ../package.json exports is applied
```

Reply:
Oh I see. But we don't get this far here because plain specifier detection doesn't catch `./`, `../`, `/` or URLs.

Reply:
Ah, may have lost track of that. Add a quick note to the exports project board just in case. This PR may not have broken it but I'd feel better if we had a test to ensure it in the future as well. :)